modules/core/04-channel/types/errors.go

```diff
@@ -51,4 +51,5 @@ var (
 	ErrInvalidUpgradeTimeout           = errorsmod.Register(SubModuleName, 35, "upgrade timeout is invalid")
 	ErrPendingInflightPackets          = errorsmod.Register(SubModuleName, 36, "pending inflight packets exist")
 	ErrUpgradeTimeoutFailed            = errorsmod.Register(SubModuleName, 37, "upgrade timeout failed")
+	ErrInvalidPruningAmount            = errorsmod.Register(SubModuleName, 38, "invalid pruning amount")
```

Comment:
I guess `ErrInvalidPruningLimit` or something like that if we're going with `limit`